packages/filesystem/src/browser/filesystem-saveable-service.ts

```diff
@@ -89,26 +89,20 @@ export class FilesystemSaveableService extends SaveableService {
      * @param target The new URI for the widget.
      * @param overwrite
      */
-    private async copyAndSave(sourceWidget: Widget & SaveableSource & Navigatable, target: URI, overwrite: boolean): Promise<void> {
-        const snapshot = sourceWidget.saveable.createSnapshot!();
-        if (!await this.fileService.exists(target)) {
-            const sourceUri = sourceWidget.getResourceUri()!;
-            if (this.fileService.canHandleResource(sourceUri)) {
-                await this.fileService.copy(sourceUri, target, { overwrite });
-            } else {
-                await this.fileService.createFile(target);
-            }
-        }
-        const targetWidget = await open(this.openerService, target, { widgetOptions: { ref: sourceWidget } });
-        const targetSaveable = Saveable.get(targetWidget);
-        if (targetWidget && targetSaveable && targetSaveable.applySnapshot) {
-            targetSaveable.applySnapshot(snapshot);
-            await sourceWidget.saveable.revert!();
-            sourceWidget.close();
-            Saveable.save(targetWidget, { formatType: FormatType.ON });
+    protected async copyAndSave(sourceWidget: Widget & SaveableSource & Navigatable, target: URI, overwrite: boolean): Promise<void> {
```

Comment:
> As far as I understand the core change of this is to use createFile triggering a CREATE event instead of copy which may trigger a COPY or MOVE event, right?

Yes, exactly 👍 

> Since it is just an internal protected method it probably is but I wanted to raise it and maybe we should actually rename the method then.

I've renamed the method. 

> Can we safely skip that call in the new behavior and simply write out the snapshot as the copied file content? Just wondering because the documentation of snapshot is simply "Creates a snapshot of the dirty state."

I was confused by this as well. The initial implementation assumed that the snapshot is effectively just a delta state of the file. Funnily enough, this implementation stems from my very first PR to Theia, see https://github.com/eclipse-theia/theia/pull/9022. However, the snapshot actually always contains the whole model of the widget in a string format. That's why we can just save it to disk.

Reply:
Love that you are actually the original author of this method :dagger: 

You are right, it is true for our `Saveable` models, i.e., they always return the whole widget model as a string. I'm just worried that this is not universally true since the documentation does not seem to suggest that this needs to be the case or maybe I'm missing something. Can you maybe point me to the VS Code behavior, if it is the same, I think we could argue that alignment is more important than backwards compatibility. I have to leave for a couple of hours now but I'll think about this a bit more and come back to this PR afterwards.

Reply:
> I'm just worried that this is not universally true since the documentation does not seem to suggest that this needs to be the case or maybe I'm missing something.

That's true. I think we can safely adjust the documentation for this though without running into too many compatibility issues.

> Can you maybe point me to the VS Code behavior, if it is the same, I think we could argue that alignment is more important than backwards compatibility.

VS Code has a pretty different model, when it comes to the "save as" operation. Given that the different types of widgets are limited in VS Code (since it's not extendable), each type of model has it's own save as handling. See [here as an example](https://github.com/microsoft/vscode/blob/88d860624c88b77c9f5af85452082aca53442949/src/vs/workbench/services/textfile/browser/textFileService.ts#L362).

Reply:
Ah yes, I can see what you mean. Thank you for the adjusted documentation. I believe with this we should be on the safer side. Still, one could argue that as a breaking change, should that be listed somewhere or is it fine as is? Both is alright with me.

Reply:
I've added this as a breaking change to the changelog.